pil-analyzer/src/evaluator.rs

```diff
@@ -821,7 +836,57 @@ impl<'a, 'b, T: FieldElement, S: SymbolLookup<'a, T>> Evaluator<'a, 'b, T, S> {
                         }
                         ta
                     });
-                    self.symbols.lookup(&poly.name, &type_args)?
+
+                    let impl_pos = type_args
+                        .as_ref()
+                        .and_then(|type_args| poly.resolved_impls.get(type_args).as_ref().copied());
+
+                    match impl_pos {
+                        Some(index) => {
+                            let local_type_args = poly
+                                .type_args
+                                .clone()
+                                .map(|ta| {
+                                    ta.into_iter()
+                                        .filter_map(|ty| {
+                                            let key = ty.to_string();
+                                            self.type_args
+                                                .get(&key)
+                                                .map(|value| (key, value.clone()))
+                                        })
+                                        .collect::<HashMap<_, _>>()
+                                })
+                                .unwrap();
+
+                            let poly_name = poly.name.clone().replace('.', "::");
```

Comment:
This is needed because traits are formatted with `.` since they (usually) just have two components while the trait function uses `::` with its three components, right?

Since we do not maintain compatibility with polygon-pil any more, it's probably time to remove the `.` in the Display function.

Can you do this in a separate (independent) PR?

Reply:
Here: https://github.com/powdr-labs/powdr/pull/1694